tests/Games/NexusMods.Games.TestFramework/AGameTest.cs

```diff
@@ -234,8 +234,12 @@ protected async Task<TemporaryPath> CreateTestArchive(IDictionary<RelativePath,
     {
         var file = TemporaryFileManager.CreateFile();
 
-        await using var stream = FileSystem.OpenFile(file.Path, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read);
-        using var zipArchive = new ZipArchive(stream, ZipArchiveMode.Create);
+        await using var stream = file.Path.Create();
+
+        // Don't put this in Create mode, because for some reason it will create broken Zips that are not prefixed
+        // with the ZIP magic number. Not sure why and I can't reproduce it in a simple test case, but if you open
+        // in create mode all your zip archives will be prefixed with 0x0000FFFF04034B50 instead of 0x04034B50.
+        using var zipArchive = new ZipArchive(stream, ZipArchiveMode.Update);
```

Comment:
Could you update the comment and link to https://github.com/dotnet/runtime/blob/23886f158cf925e13c72e661b9891df704806746/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs#L949-L956?